### ISSUE TITLE
Unexpected redirect to localhost:999

### ISSUE THREAD

**Issue by dfelgar-ae** [hashicorp-forge/hermes#71]
1. Download and build Hermes
2. Setup Google Workspace, Algolia, Google Drive
3. Copy credentials.json
4. set HERMES_WEB_GOOGLE_OAUTH2_CLIENT_ID
5. ./hermes server -config=config.hcl

First I got the error from Google: Error 400 redirect_uri_mismatch. Redirect_uri: http://localhost:9999/callback
I tried adding that redirect_uri to the credentials via https://console.cloud.google.com/. Now when I try to start up Hermes the browser tries to open "http://localhost:9999/callback" which of course doesn't work.

Any suggestions? Thanks

**Comment by johnfuller** [comment]
@dfelgar-ae what didn't work for you?

This seems to be working as designed. Note [this line](https://github.com/hashicorp-forge/hermes/blob/01a40beec9bafea71b50182c72fbe2fac2cf480f/pkg/googleworkspace/service.go#L178). Hermes starts a server on port 9999 and then saves the token once you authorize it. This worked for me, and it returned me to the screen "The token has been recorded and this window can be closed." Were you able to get that far?

**Comment by dfelgar-ae** [comment]
I installed Hermes on a Linux EC2. When I start it up it produces a link https://accounts.google.com/o/oauth2/auth?client_id=..., to which I browse on my local computer, then I get redirected to "http://localhost:9999" which doesn't work from my local computer of course.

I installed Lynx on the Hermes machine and with that browsed to the localhost:9999 link. Now I'm getting a different error. I'll see if I have authority to add this trick to your project markdown file. Thanks for your help.
